### DISCUSSION TITLE
How can I have different line-numbers format for unified/side-by-side

### DISCUSSION THREAD

**Discussion by bew** [dandavison/delta#749] [Q&A] [answered]
Hello,

I usually use the unified layout but sometimes want to enable the side-by-side layout.

The way I enable the SBS layout is with git aliases like `git -c delta.side-by-side=true diff ....`

Currently I have: (reduced to minimal)
```ini
[delta]
  features = decorations
[delta "decorations"]
  line-numbers-left-format = {nm:^5}|
  line-numbers-right-format = {np:^5}│
```
which is perfect for the unified layout, but not so great for SBS (I'd like `│{nm:^5}│` & `│{np:^5}│`)

Is it possible to have different `line-numbers-*-format` configs for when the SBS layout is active?

**Comment by dandavison** [comment, marked as answer]
Hi @bew, yes the way to achieve these sorts of conditional customizations is to use delta's "features" feature. Here's an example:

```gitconfig
[delta]
    features = my-line-numbers

[delta "my-line-numbers"]
    line-numbers = true
    line-numbers-left-format = {nm:^5}|
    line-numbers-right-format = {np:^5}│

[delta "my-side-by-side"]
    side-by-side = true
    line-numbers-left-format = │{nm:^5}│
    line-numbers-right-format = │{np:^5}│
```

Now, by default we have line numbers:
<table><tr><td><img width=400px src="https://user-images.githubusercontent.com/52205/139070000-189f8d39-5a7f-4c90-8d91-1d593efb0f83.png" alt="image" /></td></tr></table>

and we can activate side-by-side as you demonstrated: `git -c delta.features=my-side-by-side ...`

<table><tr><td><img width=400px src="https://user-images.githubusercontent.com/52205/139070275-114008eb-f595-4a4f-babd-d8df1587c9b0.png" alt="image" /></td></tr></table>

As you can imagine, specifying the precise behavior when you start adding multiple features (`features = feature-1 feature-2`), or when features themselves start defining `features`, or when you name a custom feature the same as a built-in feature (e.g. "line-numbers", "side-by-side"), gets a bit involved. But the hope is that it's easy enough to find something which does what you want, and some may find it fun to experiment.

**Reply by bew** [reply]
Working perfectly, thanks a lot!!!
`delta` is a beast and the people around are awesome ✨
